src/clients/consumer/consumer.ts

```diff
@@ -286,7 +291,15 @@ export class Consumer<Key = Buffer, Value = Buffer, HeaderKey = Buffer, HeaderVa
       return callback[kCallbackPromise]
     }
 
-    this.#fetch(options, callback)
+    clientsChannel.traceCallback(
+      this.#fetch,
+      1,
+      createDiagnosticContext({ client: this, operation: 'fetch', options }),
```

Comment:
Might make sense to have separate channels per-operation, but not a hard requirement. Just a bit more efficient if you only care about certain operations to subscribe selectively rather than filter events.